periodic neighbour related issues breaks OSX build

subj, at least at the most recent XCode 7.3.:
```
source/grid/tria_accessor.cc:1926:84: error: no viable conversion from '__map_const_iterator<__tree_const_iterator<__value_type<pair<[2 * ...]>, [...]>,
      std::__1::__tree_node<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, [...]>>' to
      '__map_const_iterator<__tree_const_iterator<__value_type<const pair<[2 * ...]>, [...]>, std::__1::__tree_node<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>,
      std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, [...]>>'
  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
                                                                                   ^
/Users/davydden/libs-sources/deal.ii/builds/fe_enriched/source/grid/tria_accessor.inst:27:17: note: in instantiation of member function 'dealii::CellAccessor<1, 1>::periodic_neighbor' requested here
 template class CellAccessor< 1 >;
                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:760:29: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const_iterator' (aka
      '__map_const_iterator<__tree_const_iterator<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >,
      std::__1::__tree_node<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, long> >') to
      'const std::__1::__map_const_iterator<std::__1::__tree_const_iterator<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned
      int>, std::__1::bitset<3> > >, std::__1::__tree_node<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>,
      std::__1::bitset<3> > >, void *> *, long> > &' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY __map_const_iterator
                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:760:29: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'const_iterator' (aka
      '__map_const_iterator<__tree_const_iterator<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >,
      std::__1::__tree_node<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, long> >') to
      'std::__1::__map_const_iterator<std::__1::__tree_const_iterator<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>,
      std::__1::bitset<3> > >, std::__1::__tree_node<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3>
      > >, void *> *, long> > &&' for 1st argument
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:785:5: note: candidate constructor not viable: no known conversion from 'const_iterator' (aka
      '__map_const_iterator<__tree_const_iterator<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >,
      std::__1::__tree_node<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, long> >') to
      'std::__1::__tree_const_iterator<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >,
      std::__1::__tree_node<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, long>'
      for 1st argument
    __map_const_iterator(_TreeIterator __i) _NOEXCEPT : __i_(__i) {}
    ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/map:787:5: note: candidate constructor not viable: no known conversion from 'const_iterator' (aka
      '__map_const_iterator<__tree_const_iterator<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >,
      std::__1::__tree_node<std::__1::__value_type<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, long> >') to
      '__map_iterator<typename __tree_const_iterator<__value_type<const pair<TriaIterator<CellAccessor<1> >, unsigned int>, pair<pair<TriaIterator<CellAccessor<1> >, unsigned int>, bitset<3> > >, __tree_node<__value_type<const pair<TriaIterator<CellAccessor<1> >, unsigned
      int>, pair<pair<TriaIterator<CellAccessor<1> >, unsigned int>, bitset<3> > >, void *> *, long>::__non_const_iterator>' (aka '__map_iterator<__tree_iterator<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>,
      std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, std::__1::__tree_node<std::__1::__value_type<const std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>,
      std::__1::pair<std::__1::pair<dealii::TriaIterator<dealii::CellAccessor<1> >, unsigned int>, std::__1::bitset<3> > >, void *> *, long> >') for 1st argument
    __map_const_iterator(__map_iterator<
    ^
```

same at
```
source/grid/tria_accessor.cc:1977:84
source/grid/tria_accessor.cc:2030:84
source/grid/tria_accessor.cc:2109:84
source/grid/tria_accessor.cc:2148:84
```

@samiiali @masterleinad : could you guys please have a look?

--- a/source/grid/tria_accessor.cc
+++ b/source/grid/tria_accessor.cc
@@ -1,6 +1,6 @@
 ﻿// ---------------------------------------------------------------------
 //
-// Copyright (C) 1998 - 2015 by the deal.II authors
+// Copyright (C) 1998 - 2016 by the deal.II authors
 //
 // This file is part of the deal.II library.
 //
@@ -1923,8 +1923,11 @@ periodic_neighbor (const unsigned int i_face) const
   typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
   typedef std::pair<cell_face_pair, std::bitset<3> > oriented_cell_face_pair;
   cell_iterator my_it(*this);
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        my_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
   // Assertion is required to check that we are actually on a periodic boundary.
   Assert (my_face_pair != this->tria->periodic_face_map.end(),
           TriaAccessorExceptions::ExcNoPeriodicNeighbor());
@@ -1974,8 +1977,10 @@ periodic_neighbor_child_on_subface (const unsigned int i_face,
   typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
   typedef std::pair<cell_face_pair, std::bitset<3> > oriented_cell_face_pair;
   cell_iterator my_it(*this);
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        my_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
   /*
    * There should be an assertion, which tells the user that this function should not be
    * used for a cell which is not located at a periodic boundary.
@@ -2027,8 +2032,10 @@ periodic_neighbor_of_coarser_periodic_neighbor (const unsigned int i_face) const
   typedef std::pair<cell_face_pair, std::bitset<3> > oriented_cell_face_pair;
   const int my_face_index = this->face_index(i_face);
   cell_iterator my_it(*this);
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        my_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
   /*
    * There should be an assertion, which tells the user that this function should not be
    * used for a cell which is not located at a periodic boundary.
@@ -2037,8 +2044,10 @@ periodic_neighbor_of_coarser_periodic_neighbor (const unsigned int i_face) const
           TriaAccessorExceptions::ExcNoPeriodicNeighbor());
   cell_iterator nb_it = my_face_pair->second.first.first;
   unsigned int face_num_of_nb = my_face_pair->second.first.second;
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator nb_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(nb_it, face_num_of_nb));
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        nb_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(nb_it, face_num_of_nb));
   /*
    * Since, we store periodic neighbors for every cell (either active or artificial or inactive)
    * the nb_face_pair should also be mapped to some cell_face pair. We assert this here.
@@ -2106,8 +2115,10 @@ CellAccessor<dim, spacedim>::periodic_neighbor_face_no (const unsigned int i_fac
   typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
   typedef std::pair<cell_face_pair, std::bitset<3> > oriented_cell_face_pair;
   cell_iterator my_it(*this);
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        my_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
   /*
    * There should be an assertion, which tells the user that this function should not be
    * called for a cell which is not located at a periodic boundary !
@@ -2145,8 +2156,10 @@ CellAccessor<dim, spacedim>::periodic_neighbor_is_coarser (const unsigned int i_
   typedef std::pair<cell_iterator, unsigned int> cell_face_pair;
   typedef std::pair<cell_face_pair, std::bitset<3> > oriented_cell_face_pair;
   cell_iterator my_it(*this);
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator my_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        my_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(my_it, i_face));
   /*
    * There should be an assertion, which tells the user that this function should not be
    * used for a cell which is not located at a periodic boundary.
@@ -2155,8 +2168,10 @@ CellAccessor<dim, spacedim>::periodic_neighbor_is_coarser (const unsigned int i_
           TriaAccessorExceptions::ExcNoPeriodicNeighbor());
   cell_iterator nb_it = my_face_pair->second.first.first;
   unsigned int face_num_of_nb = my_face_pair->second.first.second;
-  typename std::map<const cell_face_pair, oriented_cell_face_pair>::const_iterator nb_face_pair =
-    this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(nb_it, face_num_of_nb));
+  const typename std::map<std::pair<cell_iterator, unsigned int>,
+        std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > >::const_iterator
+        nb_face_pair =
+          this->tria->periodic_face_map.find(std::pair<cell_iterator, unsigned int>(nb_it, face_num_of_nb));
   /*
    * Since, we store periodic neighbors for every cell (either active or artificial or inactive)
    * the nb_face_pair should also be mapped to some cell_face pair. We assert this here.

